.github/workflows/build_and_deploy_hold.yml

```diff
@@ -0,0 +1,116 @@
+      BROWSERSTACK_ACCESS_KEY: ${{ secrets.BROWSERSTACK_ACCESS_KEY }}
+
+  call_format_branch_name:
+    uses: ./.github/workflows/format_branch_name.yml
+
+  extract_versions:
+    runs-on: ubuntu-latest
+    outputs:
+      minor_version: ${{ steps.vars.outputs.minor_version }}
+      major_version: ${{ steps.vars.outputs.major_version }}
+    steps:
+      - name: extract major and minor substrings
+        id: vars
+        run: |
+          MINOR_VERSION="$(echo "${{ github.ref_name }}" | cut -d '.' -f 1,2)"  
+          echo $MINOR_VERSION
+          echo ::set-output name=minor_version::${MINOR_VERSION}
+          MAJOR_VERSION="$(echo "${{ github.ref_name }}" | cut -d '.' -f 1)"
+          echo $MAJOR_VERSION
+          echo ::set-output name=major_version::${MAJOR_VERSION}
+
+  should_deploy_major_version:
+    runs-on: ubuntu-latest
+    outputs:
+      deploy_major_version: ${{ steps.vars.outputs.deploy_major_version }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: allow for major version deployment if the next minor version from current tag does not exist
+        id: vars
+        run: |
+          MINOR_VERSION=$(echo "${{ github.ref_name }}" | cut -d '.' -f 2) 
+          MAJOR_VERSION=$(echo "${{ github.ref_name }}" | cut -d '.' -f 1)
+          NEXT_MINOR_VERSION=$(( $MINOR_VERSION + 1 ))
+          OUTPUT=$(git tag --list "$MAJOR_VERSION.$NEXT_MINOR_VERSION.*")
+          if [ -z $OUTPUT ]
+          then
+            echo ::set-output name=deploy_major_version::true
+          fi
+
+  call_deploy_tag_version:
+    needs: 
+      - call_unit_test
+      - call_format_branch_name
+      - call_acceptance
+      - extract_versions
+      - call_misc_tests
+    uses: ./.github/workflows/deploy_hold.yml
+    with:
+      directory: ${{ github.ref_name }}
+      cache-control: 'max-age=31536000'
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+  call_deploy_major_version:
+    needs: 
+      - call_unit_test
+      - call_format_branch_name
+      - call_acceptance
+      - extract_versions
+      - call_misc_tests
+      - should_deploy_major_version
+    if: ${{ needs.should_deploy_major_version.outputs.deploy_major_version }}
+    uses: ./.github/workflows/deploy_hold.yml
+    with:
+      directory: ${{ needs.extract_versions.outputs.major_version }}
+      cache-control: 'max-age=43200'
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+  call_deploy_minor_version:
+    needs: 
+      - call_unit_test
+      - call_format_branch_name
+      - call_acceptance
+      - extract_versions
+      - call_misc_tests
+    uses: ./.github/workflows/deploy_hold.yml
+    with:
+      directory: ${{ needs.extract_versions.outputs.minor_version }}
```

Comment:
I was just about to approve but just wanted to double check, if the version is say 1.12.1, will the `minor_version` here be 'v1.12', or '12'? Because we should be deploying into the 'v1.12' directory and not the '12' directory

Reply:
major version and minor version output coming from extract_versions job would be proper strings for the deployment directory. So v1.12 and v1.12.1!